tests/test_core.py

```diff
@@ -233,51 +229,60 @@ def _validate_soundscape_and_event_audio(orig_wav_file, gen_wav_file):
 
         # Now add in trimming!
         for _ in range(5):
-            sc.generate(orig_wav_file.name, orig_jam_file.name,
-                        disable_instantiation_warnings=True,
-                        save_isolated_events=True)
-            scaper.trim(orig_wav_file.name, orig_jam_file.name,
-                        orig_wav_file.name, orig_jam_file.name,
-                        np.random.uniform(0, 5), np.random.uniform(5, 10))
-            scaper.generate_from_jams(orig_jam_file.name, gen_wav_file.name,
-                                      save_isolated_events=True)
-
-            _validate_soundscape_and_event_audio(orig_wav_file, gen_wav_file)
+            with backports.tempfile.TemporaryDirectory() as gen_events_path:
+                sc.generate(orig_wav_file.name, orig_jam_file.name,
+                            disable_instantiation_warnings=True,
+                            save_isolated_events=True)
```

Comment:
since we don't specify the output folder here for the isolated events, it'll save it to the same folder as the soundscape file, which is a NamedTemporaryFile. I'm not sure if that's a safe operation - it might be safer to create one temp folder for the "original" isolated events, and then another temp folder for the re-generated isolated events. This way we're 100% confident all isolated events are saved into legit temporary folders.